src/main/kotlin/org/rust/ide/inspections/lints/RsUnknownCrateTypesInspection.kt

```diff
@@ -0,0 +1,33 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.inspections.lints
+
+import com.intellij.psi.PsiElement
+import org.rust.ide.inspections.RsProblemsHolder
+import org.rust.lang.core.psi.RsLitExpr
+import org.rust.lang.core.psi.RsMetaItem
+import org.rust.lang.core.psi.RsVisitor
+import org.rust.lang.core.psi.ext.stringValue
+
+class RsUnknownCrateTypesInspection : RsLintInspection() {
+    override fun getLint(element: PsiElement): RsLint = RsLint.UnknownCrateTypes
+
+    override fun buildVisitor(holder: RsProblemsHolder, isOnTheFly: Boolean): RsVisitor =
+        object : RsVisitor() {
+            override fun visitLitExpr(o: RsLitExpr) {
+                val parent = o.parent as? RsMetaItem ?: return
```

Comment:
Also, I suppose, we also should check that we are in inner attribute since the compiler for following code
```rust
#[crate_type = "foo"]
fn main() {}
```
produces only
```
warning: unused attribute
 --> src/main.rs:1:1
  |
1 | #[crate_type = "foo"]
  | ^^^^^^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(unused_attributes)]` on by default

warning: crate-level attribute should be an inner attribute: add an exclamation mark: `#![foo]`
 --> src/main.rs:1:1
  |
1 | #[crate_type = "foo"]
  | ^^^^^^^^^^^^^^^^^^^^^

warning: 2 warnings emitted
```

Reply:
`org.rust.lang.core.psi.ext.RsMetaItemKt#isRootMetaItem` can also help here. If you pass `ProcessingContext` into it, it put attribute into it by `org.rust.lang.core.RsPsiPattern#META_ITEM_ATTR` key